polars/polars-core/src/chunked_array/ops/explode.rs

```diff
@@ -307,32 +307,26 @@ pub(crate) fn offsets_to_indexes(offsets: &[i64], capacity: usize) -> Vec<IdxSiz
     if offsets.is_empty() {
         return vec![];
     }
+
     let mut idx = Vec::with_capacity(capacity);
 
-    // `value_count` counts the taken values from the list values
-    // and are the same unit as `offsets`
-    // we also add the start offset as a list can be sliced
-    let mut value_count = offsets[0];
     let mut last_idx = 0;
-
-    for offset in &offsets[1..] {
-        // this get all the elements up till offsets
-        while value_count < *offset {
-            value_count += 1;
-            idx.push(last_idx)
+    for (offset_start, offset_end) in offsets.iter().zip(offsets.iter().skip(1)) {
```

Comment:
Can you replace `offsets.iter().skip(1)` with `offsets[1..].iter()` that should be faster as this removes a branch in the hot loop.

Reply:
I can go ahead and make the change, though it doesn't appear to significantly impact performance. If I'm not wrong, `offsets.iter().skip(1)` only gets called once in this function at the start of the loop, so the difference should be small, if there is one. Re-running the benchmark with this change suggests the performance difference is ±1%, which I would guess is mostly due to noise.

Reply:
No, the skip will be called every iteration and has an `unlikely` branch. The branch prediction will correctly predict the correct branch, but it can prevent future compiler optimizations to have that branch.